src/librustc/middle/dead.rs

```diff
@@ -464,20 +460,18 @@ impl<'a, 'tcx> DeadVisitor<'a, 'tcx> {
         // method of a private type is used, but the type itself is never
         // called directly.
         let impl_items = self.tcx.impl_items.borrow();
-        match self.tcx.inherent_impls.borrow().get(&self.tcx.map.local_def_id(id)) {
-            None => (),
-            Some(impl_list) => {
-                for impl_did in impl_list.iter() {
-                    for item_did in impl_items.get(impl_did).unwrap().iter() {
-                        if let Some(item_node_id) =
-                                self.tcx.map.as_local_node_id(item_did.def_id()) {
-                            if self.live_symbols.contains(&item_node_id) {
-                                return true;
-                            }
-                        }
-                    }
-                }
-            }
+        if let Some(impl_list) =
+            self.tcx.inherent_impls.borrow().get(&self.tcx.map.local_def_id(id)) {
+             for impl_did in impl_list.iter() {
```

Comment:
nit: this line and the following block are indented 5 spaces relative to the `if let` -- should be 4.